src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -364,9 +366,12 @@ private async Task<TestLspServer> CreateTestLspServerAsync(TestWorkspace workspa
             return await TestLspServer.CreateAsync(workspace, lspOptions, TestOutputLspLogger);
         }
 
-        internal TestWorkspace CreateWorkspace(InitializationOptions? options, string? workspaceKind)
+        internal TestWorkspace CreateWorkspace(
+            InitializationOptions? options, string? workspaceKind, bool mutatingLspWorkspace)
         {
-            var workspace = new TestWorkspace(Composition, workspaceKind, configurationOptions: new WorkspaceConfigurationOptions(EnableOpeningSourceGeneratedFiles: true));
+            var workspace = mutatingLspWorkspace
+                ? new LspMutatingTestWorkspace(Composition, workspaceKind, configurationOptions: new WorkspaceConfigurationOptions(EnableOpeningSourceGeneratedFiles: true))
+                : new TestWorkspace(Composition, workspaceKind, configurationOptions: new WorkspaceConfigurationOptions(EnableOpeningSourceGeneratedFiles: true));
```

Comment:
here's where we feed teh flag all the way down to, eventually creating an appropriate workspace.

Note: i don't love having these two types.  An alternative is that instead of having just a marker interface, we can have a marker interface witha  bool saying which mode you're in (fork, or mutate).  Then TestWorkspace would implement the interface, and return the particular value the test is configured for.  Thoughts @dibarbet ?

Reply:
I think I prefer the marker interface with a bool - then you could pass in the desired value to the testworkspace ctor.